File: Epic-Dining-Backend/Cafe-User-Service/src/main/java/com/example/CafeUserService/Service/OrderService.java
package com.example.CafeUserService.Service;

import com.example.CafeUserService.Model.Address;
import com.example.CafeUserService.Model.Order;

import java.util.List;

public interface OrderService {

    Order saveOrder(String userEmail, Order order);

    List<Order> getOrdersByUserId(String userEmail);

    List<Order> getAllOrders();

    boolean deleteOrder(String userEmail, String orderId);

    Address saveAddressForOrder(String userEmail, String orderId, Address address);
}

File: Epic-Dining-Frontend/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import { Order } from '../Models/order';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Address } from '../Models/address';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private http: HttpClient) { }

  endPoint = "http://localhost:9000/orders"

  saveOrder(userEmail: string, order: Order): Observable<Order> {
    return this.http.post<Order>(`${this.endPoint}/save/${userEmail}`, order);
  }

  getOrdersByUserId(userEmail: string): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.endPoint}/fetch/${userEmail}`);
  }

  getAllOrder(): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.endPoint}/`);
  }

  deleteOrder(userEmail: string, orderId: string): Observable<void> {
    return this.http.delete<void>(`${this.endPoint}/delete/${userEmail}/${orderId}`);
  }

  saveAddressForOrder(userEmail: string, orderId: string, address: Address): Observable<Address> {
    return this.http.post<Address>(`${this.endPoint}/add/${userEmail}/${orderId}/address`, address);
  }
}
